src/core/geometry/qgsgeometryutils.cpp

```diff
@@ -834,6 +834,82 @@ QgsPointV2 QgsGeometryUtils::midpoint( const QgsPointV2 &pt1, const QgsPointV2 &
   return QgsPointV2( pType, x, y, z, m );
 }
 
+double QgsGeometryUtils::slope( const QgsPointV2 &pt1, const QgsPointV2 &pt2 )
+{
+  double delta_x = pt2.x() - pt1.x();
+  double delta_y = pt2.y() - pt1.y();
+  if ( qgsDoubleNear( delta_x, 0.0 ) )
+  {
+    return INFINITY;
+  }
+  return delta_y / delta_x;
+}
+
+QVector<double> QgsGeometryUtils::coefficients( const QgsPointV2 &pt1, const QgsPointV2 &pt2 )
+{
+  QVector<double> coef;
+  if ( qgsDoubleNear( pt1.x(), pt2.x() ) )
+  {
+    coef.append( 1 );
+    coef.append( 0 );
+    coef.append( -pt1.x() );
+  }
+  else if ( qgsDoubleNear( pt1.y(), pt2.y() ) )
+  {
+    coef.append( 0 );
+    coef.append( 1 );
+    coef.append( -pt1.y() );
+  }
+  else
+  {
+    coef.append( pt1.y() - pt2.y() );
+    coef.append( pt2.x() - pt1.x() );
+    coef.append( pt1.x() * pt2.y() - pt1.y() * pt2.x() );
+  }
+
+  return coef;
+}
+
+QgsLineString* QgsGeometryUtils::perpendicularSegment( const QgsPointV2 &p, const QgsPointV2 &s1, const QgsPointV2 &s2 )
+{
+  QgsLineString *line = new QgsLineString;
```

Comment:
I'd like to see us move toward the more modern c++11 approach of using a unique_ptr here, which is only released when the object is returned